### PR TITLE
feat:session inactivity fix

### PR DESCRIPTION
**Pull request by SowmyaFyle**
### <samp>🤖 Generated by Copilot at ad9beef</samp>

This pull request enables the use of the access token from the token service for the router API service authentication in the `AuthService` and `RouterAuthService` classes. It also updates the test for the `RouterAuthService` class to use the access token.

### <samp>🤖 Generated by Copilot at ad9beef</samp>

> _We'll use the `access token` to sail the router sea_
> _We'll fetch the `token service` and post it merrily_
> _We'll heed the linting rule and comment as we please_
> _We'll pull the code together on the count of three_

### <samp>🤖 Generated by Copilot at ad9beef</samp>

*  Add and use access token for router API service authentication ([link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-54ffc766441f21f11e63851c09e17e4c6e2a3c0c282662c069608965281a258eR43-R44), [link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-54ffc766441f21f11e63851c09e17e4c6e2a3c0c282662c069608965281a258eR55), [link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-f844a6a1024ed2efa3ca2ba71f0d8975e3b1849d66a7e62c997ea61beac97049R221), [link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-ad3428039dbde906d2957e4708af02b29e8ee60add1e52caf8ef97bd067cdcf5L77-R83))
* Disable eslint rule for unknown or any return type from router API service ([link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-ad3428039dbde906d2957e4708af02b29e8ee60add1e52caf8ef97bd067cdcf5R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2184/files?diff=unified&w=0#diff-ad3428039dbde906d2957e4708af02b29e8ee60add1e52caf8ef97bd067cdcf5L77-R83))

## Clickup
https://app.clickup.com/t/85ztg262f

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -40,16 +40,20 @@ export class AuthService {
 
   newRefreshToken(token: string): Observable<ExtendedOrgUser> {
     const that = this;
+    const accessToken = from(this.tokenService.getAccessToken());
```
Comment (Julias0) [inline src/app/core/services/auth.service.ts:43]: Please to test out login and opening app post login to see if this is working properly

File: src/app/core/services/router-auth.service.spec.ts [hunk 2]
```diff
@@ -212,12 +212,16 @@ describe('RouterAuthService', () => {
...
   it('fetchAccessToken(): should fetch access token', fakeAsync(() => {
     routerApiService.post.and.returnValue(of(apiAuthRes));
+    tokenService.getAccessToken.and.returnValue(Promise.resolve(access_token));
 
     tick();
-    routerAuthService.fetchAccessToken(refresh_token).then((res) => {
-      expect(res).toEqual(apiAuthRes);
-      expect(routerApiService.post).toHaveBeenCalledOnceWith('/auth/access_token', {
-        refresh_token,
+    tokenService.getAccessToken().then((access_token) => {
```
Comment (Julias0) [inline src/app/core/services/router-auth.service.spec.ts:218]: why do we need this? We already have `access_token` here

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comment

File: src/app/core/services/auth.service.ts (hunk 1 above)
Comment (SowmyaFyle) [reply src/app/core/services/auth.service.ts:43]: tested 🍏

File: src/app/core/services/router-auth.service.spec.ts (hunk 2 above)
Comment (SowmyaFyle) [reply src/app/core/services/router-auth.service.spec.ts]: not needed - removed

**Review by Julias0 — APPROVED** [review]

**Pull request merged**